### PR TITLE
Rename ImageGPT

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR renames `ImageGPTForCausalLM` to `ImageGPTForCausalImageModeling`.

It's a long name. I know.

### CODE REVIEW

File: src/transformers/models/auto/modeling_auto.py [hunk 1]
```diff
@@ -147,7 +147,7 @@
 MODEL_WITH_LM_HEAD_MAPPING_NAMES = OrderedDict(
     [
         # Model with LM heads mapping
-        ("imagegpt", "ImageGPTForCausalLM"),
+        ("imagegpt", "ImageGPTForCausalImageModeling"),
```
Comment (NielsRogge) [inline src/transformers/models/auto/modeling_auto.py:150]: As you can see, ImageGPT is currently included in the MODEL_WITH_LM_HEAD_MAPPING_NAMES.

Should we create a MODEL_WITH_IMAGE_MODELING_HEAD_MAPPING_NAMES instead, or this is ok?

Comment (LysandreJik) [reply src/transformers/models/auto/modeling_auto.py:150]: I would put it in a separate mapping indeed!

Comment (sgugger) [reply src/transformers/models/auto/modeling_auto.py:150]: I would remove it from that mapping that is deprecated anyway, and just use another mapping `MODEL_FOR_CAUSAL_IMAGE_MODELING_MAPPING` (and MAPPING_NAMES)

**Review by sgugger — APPROVED** [review]
Thanks for renaming this model!

**Review by LysandreJik — APPROVED** [review]
LGTM, thank you @NielsRogge

**Pull request merged**